website/addons/owncloud/model.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+from urllib2 import HTTPError
+import logging
+from modularodm import fields
+from framework.auth import Auth
+from website.addons.base import exceptions
+from website.addons.base import StorageAddonBase
+from website.addons.base import (
+    AddonOAuthNodeSettingsBase, AddonOAuthUserSettingsBase,
+)
+from website.addons.owncloud.serializer import OwnCloudSerializer
+from website.addons.owncloud.utils import (
+    ExternalAccountConverter, OwnCloudNodeLogger
+)
+
+logger = logging.getLogger(__name__)
+
+class OwnCloudProvider(object):
+    """An alternative to `ExternalProvider` not tied to OAuth"""
+
+    name = 'ownCloud'
+    short_name = 'owncloud'
+    serializer = OwnCloudSerializer
+
+    def __init__(self, account=None):
+        super(OwnCloudProvider, self).__init__()
+        self.account = account
+
+    def __repr__(self):
+        return '<{name}: {status}>'.format(
+            name=self.__class__.__name__,
+            status=self.account.display_name if self.account else 'anonymous'
+        )
+
+class AddonOwnCloudUserSettings(AddonOAuthUserSettingsBase):
+    oauth_provider = OwnCloudProvider
+    serializer = OwnCloudSerializer
+
+class AddonOwnCloudNodeSettings(StorageAddonBase, AddonOAuthNodeSettingsBase):
+    oauth_provider = OwnCloudProvider
+    serializer = OwnCloudSerializer
+
+    folder_name = fields.StringField()
+
+    @property
+    def complete(self):
```

Comment:
This should probably be using the superclass' implementation -- it looks like it's trying to do the work of the `configured` property, currently.